test/unit/fonts_spec.js

```diff
@@ -83,7 +83,7 @@ var checkProblematicCharRanges = function checkProblematicCharRanges() {
 
 describe('Fonts', function() {
   it('checkProblematicCharRanges', function() {
-    var EXPECTED_PERCENTAGE = 45;
+    var EXPECTED_PERCENTAGE = 100;
```

Comment:
Is it strictly necessary to make this a full 100 percent, for the tests to pass with this patch?

The reason that I'm asking is that with this change, there's nothing stopping us from adding enough entries to `ProblematicCharRanges` that there'd be no room over for re-mapping glyphs that would otherwise end up at a previously mapped location. Basically, wouldn't we always hit https://github.com/mozilla/pdf.js/blob/518da6c6a9df7e388e6bc4523f21f4d2044b8459/src/core/fonts.js#L787-L790 in that case, where as if we keep the value lower than 100 this test-case would ensure that there's atleast *some* space left in the PUA?

Reply:
I think it's now at 70%. I could lower it, but it seems odd to have a test that we have to change every time the ranges change. We would only run out of space in the PUA if the font had tons of values without unicode values and lots of values with problematic unicodes (or a very large symbolic font).

Reply:
Sure, let's just do this for now!
Hopefully, we'd be able to (eventually) replace the current black-list approach with a white-list instead.